lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1386,6 +1386,26 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     return b.create<math::SqrtOp>(loc, payloadArgs[0]);
   if (isa<AtenRsqrtOp>(op))
     return b.create<math::RsqrtOp>(loc, payloadArgs[0]);
+  if (auto bitwiseAndTensor = dyn_cast<AtenBitwiseAndTensorOp>(op)) {
+    if (bitwiseAndTensor.getType()
+            .cast<ValueTensorType>()
+            .getDtype()
+            .isa<mlir::FloatType>()) {
+      bitwiseAndTensor.emitError(
+          "Bitwise_And does not support floating point dtype");
+      return nullptr;
+    }
+    Value lhs = payloadArgs[0];
+    Value rhs = payloadArgs[1];
+    if (lhs.getType() != rhs.getType()) {
```

Comment:
this `if` can be removed as `convertScalarToDtype` handle the cases no conversion is happening. It's a common logic among all the ops so it's better to handle this in the common api rather than each op checking this separately.

Reply:
I think it's not entirely common among multiple ops; in the case of some ops, it converts one operand, in some cases two, and in some cases three depending on the number of input operands of op. Also, in some cases, it only derives the dtype but doesn't do the conversion. If we make the function such that, we have to call it for each operand; then it's better not to move it to a common API. So, I think it's better to leave them as they are as of now.

Reply:
I wasn't being clear in the previous common. I meant the `if (lhs.getType() != rhs.getType()) {` is redundant because it's handled by `convertScalarToDtype` already. `convertScalarToDtype` is the common API.

Reply:
Okay, I misunderstood your comment.